project/src/backend/sdl/SDLGamepad.h

```diff
@@ -11,13 +11,31 @@ namespace lime {
 
 
 	class SDLGamepad {
-
 		public:
+			SDL_GameController *gameController = nullptr;
+			
+			SDLGamepad() {}
+			SDLGamepad(SDL_GameController *_gameController) : gameController(_gameController) {}
+			
+			~SDLGamepad() {
+				// Close controller if opened
+				if (gameController != nullptr)
+					SDL_GameControllerClose(gameController);
+			}
+
+			void Rumble(int duration, double largeStrength, double smallStrength);
 
 			static bool Connect (int deviceID);
 			static int GetInstanceID (int deviceID);
 			static bool Disconnect (int id);
 
+			SDLGamepad &operator=(SDLGamepad &&other)
+			{
+				SDL_GameController *temp = gameController;
+				gameController = other.gameController;
+				other.gameController = temp;
+				return *this;
+			}
```

Comment:
Wouldn't it be clearer to define a `Swap()` function instead of overriding an operator?

Reply:
was this the final thing that needed to be addressed for this PR?

Reply:
I don't know about "final" or otherwise. This was just something I was curious about.

Reply:
fwiw, op overloading does seem to make, otherwise seemingly straightforward code have oddly obfuscated behavior